### PR TITLE
test: add testing infrastructure

### PR DESCRIPTION
**Pull request by tcrasset**

### CODE REVIEW

File: avatars/client_test.py [hunk 1]
```diff
@@ -50,3 +39,44 @@ def test_url_is_rejected_if_it_contains_quotes(base_url: str) -> None:
...
+    The handler returns an empty 200 response by default.
+    Consider overriding it with a custom handler for more complex tests.
+    """
+    if handler is None:
+        handler = lambda request: httpx.Response(200, json={})
+
+    transport = httpx.MockTransport(handler)
+    http_client = httpx.Client(base_url="http://localhost:8000", transport=transport)
+    return ApiClient(
+        base_url="http://localhost:8000",
```
Comment (tas17) [inline avatars/client_test.py:61]: nit: base url could be a constant on the top of the file

File: avatars/client_test.py [hunk 2]
```diff
@@ -50,3 +39,44 @@ def test_url_is_rejected_if_it_contains_quotes(base_url: str) -> None:
...
     with pytest.raises(ValueError, match="not to contain quotes") as exc_info:
         api_client = ApiClient(base_url=base_url)
+
+
+import httpx
+
+RequestHandle = Callable[[httpx.Request], httpx.Response]
+
+
+def api_client_factory(handler: Optional[RequestHandle] = None) -> ApiClient:
```
Comment (tas17) [inline avatars/client_test.py:49]: Why do you have 2 `api_client_factory`?

Comment (tas17) [reply avatars/client_test.py:49]: All of this is a wrong copy paste no?
(import in the wrong place too)

**Review by tas17 — APPROVED** [review]
Just mind the duplicated `api_client_factory` before merging.
IMO it's not necessary to have the
```
# Arrange
# Act
# Assert
```
But why not

File: pyproject.toml [hunk 3]
```diff
@@ -36,6 +36,7 @@ jupytext = "^1.14.2"
 nbconvert = "^7.2.6"
 flake8 = "^6.0.0"
 pytest-cov = "^4.1.0"
+rich = "^13.7.0"
```
Comment (jpetot) [inline pyproject.toml:39]: what is it for?

**Review by jpetot — APPROVED** [review]
looks good

File: avatars/client_test.py (hunk 1 above)
Comment (tcrasset) [reply avatars/client_test.py:61]: It's not needed, I could have written `"http://blabla"`, as we are not actually calling the API but using a `MockTransport`

File: avatars/client_test.py (hunk 2 above)
Comment (tcrasset) [reply avatars/client_test.py:49]: yes

File: pyproject.toml (hunk 3 above)
Comment (tcrasset) [reply pyproject.toml:39]: Don't remember :shrug:

Comment (tcrasset) [reply pyproject.toml]: Removed it

**Comment by tcrasset** [comment]
Will fix the doctest in api

**Pull request closed without merging**